tests/test_binarized_mnist.py

```diff
@@ -40,6 +40,16 @@ def test_binarized_mnist_test():
     mnist_test.close(handle)
 
 
+def test_binarized_mnist_axes():
+    skip_if_not_available(datasets=['binarized_mnist.hdf5'])
+
+    dataset = BinarizedMNIST('train', load_in_memory=False)
+    assert all(
+        dim_label == label for (dim_label, label) in
+        zip(dataset.axis_label_dict['features'],
+            ('batch', 'channel', 'axis_0', 'axis_1')))
```

Comment:
`zip` will stop at the shortest sequence, so the test would pass if `dataset.axis_label_dict['features']` was an empty list or tuple.